src/main/java/com/es/phoneshop/utils/DataParser.java

```diff
@@ -0,0 +1,25 @@
+package com.es.phoneshop.utils;
+
+import com.es.phoneshop.exception.ParseToIntegerException;
+
+import javax.servlet.http.HttpServletRequest;
+import java.text.NumberFormat;
+import java.text.ParseException;
+
+public class DataParser {
```

Comment:
I would rather make this class *Utils like, with public static methods. Just like you did with ErrorHandler. It also ok to change ErrorHandler to be like this class imo. Because you use different approaches in the project for the same kind of things.